renderer/components/Editor.tsx

```diff
@@ -260,12 +269,68 @@ export class Editor extends React.Component<Props, State> {
     }
   }
 
+  shift = (number: number, shift: number) => {
+    return number * Math.pow(2, shift);
+  }
+
   toggleConsole = () => {
     this.props.toggleConsole();
     // Resize since the console overlaps with the editor, but enough time for console changes
     setTimeout(() => this.onWindowResize(), 0.01);
   }
 
+  // toggle keyboard control and add/remove listening for key presses to control robot
+  toggleKeyboardControl = () => {
+    
+    
+    this.setState({keyboardControl: !this.state.keyboardControl})
+    
+    if (!this.state.keyboardControl) {
+      window.addEventListener('keydown', this.turnCharacterOn, { passive: true});
+      window.addEventListener('keyup', this.turnCharacterOff, { passive: true});
+      this.setState({buttonStyle: "info"});
+    } else {
+      window.removeEventListener('keydown', this.turnCharacterOn);
+      window.removeEventListener('keyup', this.turnCharacterOff);
+      this.setState({buttonStyle: 'default'});
+      this.setState({bitmap: 0});
+      this.props.onUpdateBitmap(this.state.bitmap);
+    }
+  }
+  updateBitmap = (currentCharacter: string, characterBool: boolean ) => {
+    const keyboardNum = KeyboardButtons[currentCharacter];
+    let map: number = this.state.bitmap;
+    
+    const shift = this.shift(1, keyboardNum);
+    const maxInt32Bits = 2147483648; // 2^31
```

Comment:
Usually constants the convention we follow is all uppercase, so `MAX_INT32_BITS`